doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,675 @@
+at the end of the description (before the dot). The exact form of the
+description in this case would be "Description of the arg (default is X).". In
+some cases it may be useful to explain what the default argument means, which
+can be added after a comma "Description of the arg (default is -1, which means
+all cpus).".
+
+**Good:**
+
+.. code-block:: python
+
+    class Series:
+        def plot(self, kind, color='blue', **kwargs):
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Parameters
+            ----------
+            kind : str
+                Kind of matplotlib plot.
+            color : str
+                Color name or rgb code (default is 'blue').
+            **kwargs
+                These parameters will be passed to the matplotlib plotting
+                function.
+            """
+            pass
+
+**Bad:**
+
+.. code-block:: python
+
+    class Series:
+        def plot(self, kind, **kwargs):
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Note the blank line between the parameters title and the first
+            parameter. Also, not that after the name of the parameter `kind`
+            and before the colo, a space is missing.
+
+            Also, note that the parameter descriptions do not start with a
+            capital letter, and do not finish with a dot.
+
+            Finally, the `**kwargs` parameter is missing.
+
+            Parameters
+            ----------
+
+            kind: str
+                kind of matplotlib plot
+            """
+            pass
+
+.. _docstring.parameter_types:
+
+Parameter types
+^^^^^^^^^^^^^^^
+
+When specifying the parameter types, Python built-in data types can be used
+directly:
+
+- int
+- float
+- str
+- bool
+
+For complex types, define the subtypes:
+
+- list of [int]
+- dict of {str : int}
+- tuple of (str, int, int)
+- set of {str}
+
+In case there are just a set of values allowed, list them in curly brackets
+and separated by commas (followed by a space). If one of them is the default
+value of a keyword argument, it should be listed first.:
+
+- {0, 10, 25}
+- {'simple', 'advanced'}
+
+If the type is defined in a Python module, the module must be specified:
+
+- datetime.date
+- datetime.datetime
+- decimal.Decimal
+
+If the type is in a package, the module must be also specified:
+
+- numpy.ndarray
+- scipy.sparse.coo_matrix
+
+If the type is a pandas type, also specify pandas except for Series and
+DataFrame:
+
+- Series
+- DataFrame
+- pandas.Index
+- pandas.Categorical
+- pandas.SparseArray
+
+If the exact type is not relevant, but must be compatible with a numpy
+array, array-like can be specified. If Any type that can be iterated is
+accepted, iterable can be used:
+
+- array-like
+- iterable
```

Comment:
This might be subtle. For instance, ``pd.Series(i for i in range(3))`` works, but it is not undocumented. However, I don't think we want to _replace_ "array-like" with "iterable": probably have both, although they are theoretically redundant.